### PR TITLE
SIMD-0207: Raise block limit to 50M

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- https://github.com/solana-foundation/solana-improvement-documents/pull/207

#### Summary of Changes
- Add feature to the feature set for raising block limits
- Clean up old limit justification(s)
- Add function to get new limits for SIMD-0207
- Add feature-activation path for SIMD-0207

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
Small patch to raise limits for https://github.com/solana-foundation/solana-improvement-documents/pull/207.

There are 34 lines of test, so "functional" patch is small, and I think should be backported so we can raise chain capacity slightly at a much earlier date.

**Review by tao-stones — DISMISSED** [review]
lgtm

File: cost-model/src/block_cost_limits.rs [hunk 1]
```diff
@@ -27,33 +19,35 @@ pub const WRITE_LOCK_UNITS: u64 = COMPUTE_UNIT_TO_US_RATIO * 10;
...
 /// accumulated by Transactions added to it; A transaction's compute units are
 /// calculated by cost_model, based on transaction's signatures, write locks,
 /// data size and built-in and SBF instructions.
-pub const MAX_BLOCK_UNITS: u64 =
-    MAX_BLOCK_REPLAY_TIME_US * COMPUTE_UNIT_TO_US_RATIO * MAX_CONCURRENCY;
-
-#[cfg(test)]
-static_assertions::const_assert_eq!(MAX_BLOCK_UNITS, 48_000_000);
+pub const MAX_BLOCK_UNITS: u64 = 48_000_000;
+pub const MAX_BLOCK_UNITS_SIMD_0207: u64 = 50_000_000;
```
Comment (brooksprumo) [inline cost-model/src/block_cost_limits.rs]: I get the desire to simplify the previous const declarations, but it does make this PR a bit larger than necessary. IMO these simplifications can be in their own PR that lands first.

**Review by brooksprumo — COMMENTED** [review]
Code looks good to me

File: cost-model/src/block_cost_limits.rs (hunk 1 above)
Comment (apfitzge) [reply cost-model/src/block_cost_limits.rs]: https://github.com/anza-xyz/agave/pull/4065

**Review by brooksprumo — APPROVED** [review]
